yt_dlp/extractor/sejmpl.py

```diff
@@ -0,0 +1,124 @@
+from .common import InfoExtractor
+from .redge import RedCDNLivxIE
+from ..utils import (
+    clean_html,
+    js_to_json,
+    strip_or_none,
+    traverse_obj,
+)
+
+import datetime
+
+
+class SejmIE(InfoExtractor):
+    _VALID_URL = (
+        r'https?://(?:www\.)?sejm\.gov\.pl/Sejm(?P<term>\d+)\.nsf/transmisje(?:_arch)?\.xsp(?:\?[^#]*)?#(?P<id>[\dA-F]+)',
+        r'https?://(?:www\.)?sejm\.gov\.pl/Sejm(?P<term>\d+)\.nsf/transmisje(?:_arch)?\.xsp\?(?:[^#]+&)?unid=(?P<id>[\dA-F]+)',
+        r'https?://sejm-embed\.redcdn\.pl/Sejm(?P<term>\d+)\.nsf/VideoFrame\.xsp/(?P<id>[\dA-F]+)',
+    )
+    IE_NAME = 'sejm'
+
+    _TESTS = [{
+        # multiple cameras, polish SL iterpreter
+        'url': 'https://www.sejm.gov.pl/Sejm10.nsf/transmisje_arch.xsp#6181EF1AD9CEEBB5C1258A6D006452B5',
+        'info_dict': {
+            'id': '6181EF1AD9CEEBB5C1258A6D006452B5',
+            'title': '1. posiedzenie Sejmu X kadencji',
+            'duration': 20145,
+            'live_status': 'was_live',
+            'location': 'Sala Posiedzeń',
+        },
+        'playlist_count': 5,
+    }, {
+        'url': 'https://www.sejm.gov.pl/Sejm10.nsf/transmisje.xsp?unid=7797B441DF1ACC35C1258A75002EB583',
+        'info_dict': {
+            'id': '7797B441DF1ACC35C1258A75002EB583',
+            'title': 'Konferencja prasowa marszałka Sejmu Szymona Hołowni',
+            'duration': 1972,
+            'live_status': 'was_live',
+            'location': 'hall przed Salą Kolumnową, bud. C',
+        },
+        'playlist_count': 1,
+    }, {
+        'url': 'https://sejm-embed.redcdn.pl/Sejm10.nsf/VideoFrame.xsp/FED58EABB97FBD53C1258A7400386492',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        term, video_id = self._match_valid_url(url).group('term', 'id')
+        frame = self._download_webpage(
+            f'https://sejm-embed.redcdn.pl/Sejm{term}.nsf/VideoFrame.xsp/{video_id}',
+            video_id, headers={
+                'Referer': f'https://www.sejm.gov.pl/Sejm{term}.nsf/transmisje_arch.xsp',
+            })
+        # despite it says "transmisje_arch", it works for live streams too!
+        data = self._download_json(
+            f'https://www.sejm.gov.pl/Sejm{term}.nsf/transmisje_arch.xsp/json/{video_id}',
+            video_id, headers={
+                'Referer': f'https://www.sejm.gov.pl/Sejm{term}.nsf/transmisje_arch.xsp',
+            })
```

Comment:
I don't think the referer is needed, so maybe just
```suggestion
        frame = self._download_webpage(
            f'https://sejm-embed.redcdn.pl/Sejm{term}.nsf/VideoFrame.xsp/{video_id}', video_id)
        data = self._download_json(
            f'https://www.sejm.gov.pl/Sejm{term}.nsf/transmisje_arch.xsp/json/{video_id}', video_id)
```